mlx/backend/common/primitives.cpp

```diff
@@ -203,6 +203,17 @@ void Concatenate::eval(const std::vector<array>& inputs, array& out) {
   }
 }
 
+void Conjugate::eval(const std::vector<array>& inputs, array& out) {
+  assert(inputs.size() == 1);
+  const auto& in = inputs[0];
+  if (out.dtype() == complex64) {
+    unary_fp(in, out, detail::Conjugate());
+  } else {
+    throw std::invalid_argument(
+        "[conjugate] conjugate must be called on complex input.");
+  }
```

Comment:
Numpy allows any type. Maybe we should do the same for consistency. The conjugate of a non complex type is just a buffer copy I think (see e.g. abs on an unsigned type)